sqlmesh/core/context.py

```diff
@@ -594,17 +619,25 @@ def standalone_audits(self) -> MappingProxyType[str, StandaloneAudit]:
         return MappingProxyType(self._standalone_audits)
 
     @property
-    def snapshots(self) -> t.Dict[str, Snapshot]:
+    def snapshots(self) -> t.Iterable[Snapshot]:
         """Generates and returns snapshots based on models registered in this context.
 
-        If one of the snapshots has been previosly stored in the persisted state, the stored
+        If one of the snapshots has been previously stored in the persisted state, the stored
         instance will be returned.
         """
-        return self._snapshots()
+        return self._snapshots().values()
+
+    @property
+    def _node_fqn_to_snapshot(self) -> t.Dict[str, Snapshot]:
+        return {s.fqn: s for s in self.snapshots}
```

Comment:
this is really expensive, basically, if you do a for loop now on get_snapshot, it's going to be like n^2

Reply:
Updated. Also worth mentioning that in main we call `self.snapshots` in `get_snapshot` and that has multiple state sync calls in it: https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/context.py#L1347-L1417

So I think we should generally be cautious about looping on `get_snapshot`.